src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,49 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleFormSubmit }) {
+    super({ popupSelector });
+    this._popSelector = document.querySelector(popupSelector);
+    this._deletebutton = this._popupElement.querySelector(
+      ".card__delete-button"
+    );
+    // this._deleteform = this._popupElement.querySelector("#modal-delete-card");
+    this._handleFormSubmit = handleFormSubmit;
+    this._isSubmitted = false;
+    this._currentCard = null;
+    this._cardId = null;
+  }
+
+  open(currentCard, cardId) {
+    this._currentCard = currentCard;
+    this._cardId = cardId;
+    super.open();
+  }
+
+  close() {
+    super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.